QA_Requests/MultiSelectPerf/generate_data_capture_readings.py

```diff
@@ -0,0 +1,83 @@
+import csv
+import os
+import names
+import pytest
+
+from QA_Requests.MultiSelectPerf.hq_workflows import AppCreationPage
+from QA_Requests.MultiSelectPerf.import_data_using_api import post_import_cases_from_excel
+
+
+def generate_path(output_path):
+    if not os.path.isdir(output_path):
+        print('Path not present')
+        os.mkdir(output_path)
+
+
+def generate_import_file(case_load_per_case_type, case_type_name, num_prop):
+    OUTPUT_PATH = "OutputFiles/"
+    BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+    generate_path(os.path.abspath(os.path.join(BASE_DIR, OUTPUT_PATH)))
+    file = os.path.abspath(os.path.join(BASE_DIR, OUTPUT_PATH + f'{str(case_type_name)}.csv'))
+    col_list = ["caseid", "name"] + ['prop{}p'.format(i) for i in range(0, num_prop)]
+    with open(file, 'w', newline="") as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=col_list)
+        writer.writeheader()
+        for cases in range(0, int(case_load_per_case_type)):
+            value_list = ["", names.get_full_name()] + ['val{}'.format(i) for i in range(0, num_prop)]
+            dic = dict(zip(col_list, value_list))
+            writer.writerow(dic)
+        return file
+
+
+"""This function is a one time script to create desired load on a domain. Call the function to run it"""
+
+
+def multi():
+    domain_case_load = 100
+    case_types_and_properties = {"case1": 50, "case2": 100, "case3": 200}
+    case_load_per_case_type = domain_case_load / len(case_types_and_properties)
+    for case_type, prop_count in case_types_and_properties.items():
+        file = generate_import_file(case_load_per_case_type, case_type, prop_count)
+        print(case_type, prop_count, "generated")
+        post_import_cases_from_excel(file, case_type)
+        print(file, case_type, "uploaded")
+
+
+"""This function is a one time script to pass properties into forms depepending on user defined load"""
+
+# def test_pass_prop_to_form(driver):
+#     print("Add property to form for all case types")
+#     create = AppCreationPage(driver)
+#     form = "https://www.commcarehq.org/a/casesearch/apps/view/e2373071215846318db147ca6ea88e69/form/c9260fe2bf244099b067556d73c111a4/source/#form/here_are_your_cases/item/prop_name_prop98p"
+#     create.pass_prop_to_case_list(form, 101, 200)
+
+
+"""Capture sync time"""
+
+
+def test_first_sync(driver):
+    create = AppCreationPage(driver)
```

Comment:
is this the main test module?

Reply:
`test_first_sync` and both `test_multiselect_perf` are required test modules. Sync needs to run before the loop that runs in `test_multiselect_perf` every time, so thy are two different functions that are collected once.